### ISSUE TITLE
Undefined method `schema'

### ISSUE THREAD

**Issue by mauricioklein** [rom-rb/rom-sql#81]
I'm receiving a `undefined method 'schema'` in my relation classes.
Example:

```ruby
class Element < ROM::Relation[:sql]
   register_as :element

   schema(:element) do
      associations do
         has_one :unit
      end
   end
end
```

Replacing `schema` by `dataset` works well:

```ruby
class Element < ROM::Relation[:sql]
   register_as :element
   dataset :element
end
```

However, with dataset, I can't (or at least don't know) how to setup the relation.

Would appreciate any support.
Thanks!

**Comment by flash-gordon** [comment]
@mauricioklein can you provide a stacktrace and the gem version? I wonder where exactly this NoMethodError occurs

**Comment by mauricioklein** [comment]
Hello @flash-gordon !
Thanks for the quick answer!

Here's the stack trace:
```ruby
/home/klein/workspace/jcc/lib/persistence/relations/elements.rb:3:in `<class:Element>': undefined method `schema' for Element:Class (NoMethodError)
	from /home/klein/workspace/jo-api/lib/persistence/relations/elements.rb:1:in `<top (required)>'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/rom-ed18a8299198/lib/rom/setup/auto_registration.rb:39:in `require'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/rom-ed18a8299198/lib/rom/setup/auto_registration.rb:39:in `block in load_entities'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/rom-ed18a8299198/lib/rom/setup/auto_registration.rb:38:in `map'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/rom-ed18a8299198/lib/rom/setup/auto_registration.rb:38:in `load_entities'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/rom-ed18a8299198/lib/rom/setup/auto_registration.rb:24:in `relations'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/rom-ed18a8299198/lib/rom/setup.rb:50:in `auto_registration'
	from /home/klein/workspace/jo-api/core/boot/rom.rb:19:in `block (2 levels) in <top (required)>'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/dry-component-2708e0e6484e/lib/dry/component/container.rb:51:in `call'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/dry-component-2708e0e6484e/lib/dry/component/container.rb:51:in `block in finalize'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/dry-component-2708e0e6484e/lib/dry/component/container.rb:110:in `call'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/dry-component-2708e0e6484e/lib/dry/component/container.rb:110:in `block in boot!'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/dry-component-2708e0e6484e/lib/dry/component/container.rb:109:in `tap'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/dry-component-2708e0e6484e/lib/dry/component/container.rb:109:in `boot!'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/dry-component-2708e0e6484e/lib/dry/component/container.rb:64:in `block in finalize!'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/dry-component-2708e0e6484e/lib/dry/component/container.rb:63:in `each'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bundler/gems/dry-component-2708e0e6484e/lib/dry/component/container.rb:63:in `finalize!'
	from /home/klein/workspace/jo-api/core/boot.rb:15:in `<top (required)>'
	from /home/klein/workspace/jo-api/config.ru:1:in `require_relative'
	from /home/klein/workspace/jo-api/config.ru:1:in `block in <main>'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/gems/rack-1.6.4/lib/rack/builder.rb:55:in `instance_eval'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/gems/rack-1.6.4/lib/rack/builder.rb:55:in `initialize'
	from /home/klein/workspace/jo-api/config.ru:in `new'
	from /home/klein/workspace/jo-api/config.ru:in `<main>'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/gems/rack-1.6.4/lib/rack/builder.rb:49:in `eval'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/gems/rack-1.6.4/lib/rack/builder.rb:49:in `new_from_string'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/gems/rack-1.6.4/lib/rack/builder.rb:40:in `parse_file'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/gems/rack-1.6.4/lib/rack/server.rb:299:in `build_app_and_options_from_config'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/gems/rack-1.6.4/lib/rack/server.rb:208:in `app'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/gems/rack-1.6.4/lib/rack/server.rb:336:in `wrapped_app'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/gems/rack-1.6.4/lib/rack/server.rb:272:in `start'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/gems/rack-1.6.4/lib/rack/server.rb:147:in `start'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/gems/rack-1.6.4/bin/rackup:4:in `<top (required)>'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bin/rackup:23:in `load'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bin/rackup:23:in `<main>'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bin/ruby_executable_hooks:15:in `eval'
	from /home/klein/.rvm/gems/ruby-2.2.3@jo-api/bin/ruby_executable_hooks:15:in `<main>'
```

And here is Gemfile.lock dump:

```ruby
GIT
  remote: git://github.com/dryrb/dry-component.git
  revision: 2708e0e6484e62f26ab0282479bf6801931ea1d1
  branch: master
  specs:
    dry-component (0.0.2)
      dry-auto_inject (~> 0.1)
      dry-configurable (~> 0.1)
      dry-container (~> 0.2, >= 0.2.7)
      inflecto (>= 0.0.2)

GIT
  remote: git://github.com/dryrb/dry-validation.git
  revision: 6447302f3b53766b29f29230831890a5cc3822e0
  branch: master
  specs:
    dry-validation (0.6.0)
      dry-configurable (~> 0.1, >= 0.1.3)
      dry-container (~> 0.2, >= 0.2.8)
      dry-data (~> 0.5, >= 0.5.0)
      dry-equalizer (~> 0.2)
      dry-logic (~> 0.1, >= 0.1.4)

GIT
  remote: git://github.com/rom-rb/rom-mapper.git
  revision: ae74ad120fb56a0b8944235c5ccd4e9dd9b57b58
  branch: master
  specs:
    rom-mapper (0.3.0)
      dry-equalizer (~> 0.2)
      rom-support (~> 1.0.0)
      transproc (~> 0.4.0)

GIT
  remote: git://github.com/rom-rb/rom-repository.git
  revision: 65d2f37ba8a30f453255333cd4821592ecfcb645
  branch: master
  specs:
    rom-repository (0.2.0)
      anima (~> 0.2, >= 0.2)
      rom (~> 1.0.0)
      rom-mapper (~> 0.3.0)
      rom-support (~> 1.0.0)

GIT
  remote: git://github.com/rom-rb/rom-sql.git
  revision: dbb538c723a0f1a5f7296ff6694b86dd9ee69b0b
  branch: master
  specs:
    rom-sql (0.7.0)
      dry-equalizer (~> 0.2)
      rom (~> 1.0.0)
      sequel (~> 4.18)

GIT
  remote: git://github.com/rom-rb/rom-support.git
  revision: 3018f392c0378354e06ba912e5582749d15c6030
  branch: master
  specs:
    rom-support (1.0.0)
      dry-equalizer (~> 0.2)
      transproc (~> 0.4.0)
      wisper (~> 1.6, >= 1.6.0)

GIT
  remote: git://github.com/rom-rb/rom.git
  revision: ed18a8299198a7f6dd73d146807ab501b6b980b8
  branch: master
  specs:
    rom (1.0.0)
      dry-equalizer (~> 0.2)
      rom-mapper (~> 0.3.0)
      rom-support (~> 1.0.0)

GIT
  remote: git://github.com/solnic/rodakase.git
  revision: fbf98dde3da2aa853934f303a4c1b40a54279afa
  branch: master
  specs:
    rodakase (0.0.1)
      dry-component (~> 0)
      dry-configurable (~> 0.1)
      dry-equalizer (~> 0.2)
      inflecto (~> 0)
      roda (~> 2.7)
      tilt (~> 2.0)
      transflow (~> 0.3)

GIT
  remote: git://github.com/solnic/transproc.git
  revision: ef7cbef5d22fffd53beadf5c28abdb4975a02c45
  branch: master
  specs:
    transproc (0.4.0)

GEM
  remote: https://rubygems.org/
  specs:
    abstract_type (0.0.7)
    activesupport (4.2.5.1)
      i18n (~> 0.7)
      json (~> 1.7, >= 1.7.7)
      minitest (~> 5.1)
      thread_safe (~> 0.3, >= 0.3.4)
      tzinfo (~> 1.1)
    adamantium (0.2.0)
      ice_nine (~> 0.11.0)
      memoizable (~> 0.4.0)
    addressable (2.4.0)
    analytics-ruby (2.0.13)
    anima (0.3.0)
      abstract_type (~> 0.0.7)
      adamantium (~> 0.2)
      equalizer (~> 0.0.11)
    bcrypt (3.1.10)
    bcrypt-ruby (3.1.5)
      bcrypt (>= 3.1.3)
    builder (3.2.2)
    byebug (8.2.2)
    capybara (2.6.2)
      addressable
      mime-types (>= 1.16)
      nokogiri (>= 1.3.3)
      rack (>= 1.0.0)
      rack-test (>= 0.5.4)
      xpath (~> 2.0)
    coderay (1.1.0)
    database_cleaner (1.5.1)
    diff-lcs (1.2.5)
    dotenv (2.1.1)
    dry-auto_inject (0.1.0)
    dry-configurable (0.1.3)
      thread_safe
    dry-container (0.2.8)
      dry-configurable (~> 0.1, >= 0.1.3)
      thread_safe
    dry-data (0.5.1)
      dry-configurable (~> 0.1)
      dry-container (~> 0.2)
      dry-equalizer (~> 0.2)
      dry-logic (~> 0.1)
      inflecto (~> 0.0.0, >= 0.0.2)
      kleisli (~> 0.2)
      thread_safe (~> 0.3)
    dry-equalizer (0.2.0)
    dry-logic (0.1.4)
      dry-container (~> 0.2, >= 0.2.6)
      dry-equalizer (~> 0.2)
    dry-pipeline (0.0.1)
    equalizer (0.0.11)
    excon (0.49.0)
    faraday (0.9.2)
      multipart-post (>= 1.2, < 3)
    faraday_middleware (0.10.0)
      faraday (>= 0.7.4, < 0.10)
    fog-aws (0.9.2)
      fog-core (~> 1.27)
      fog-json (~> 1.0)
      fog-xml (~> 0.1)
      ipaddress (~> 0.8)
    fog-core (1.37.0)
      builder
      excon (~> 0.45)
      formatador (~> 0.2)
    fog-json (1.0.2)
      fog-core (~> 1.0)
      multi_json (~> 1.10)
    fog-xml (0.1.2)
      fog-core
      nokogiri (~> 1.5, >= 1.5.11)
    formatador (0.2.5)
    i18n (0.7.0)
    ice_nine (0.11.2)
    inflecto (0.0.2)
    ipaddress (0.8.3)
    json (1.8.3)
    kleisli (0.2.7)
    levenshtein (0.2.2)
    memoizable (0.4.2)
      thread_safe (~> 0.3, >= 0.3.1)
    method_source (0.8.2)
    mime-types (3.0)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2015.1120)
    mini_portile2 (2.0.0)
    minitest (5.8.4)
    multi_json (1.11.2)
    multipart-post (2.0.0)
    mysql2 (0.4.2)
    newrelic_rpm (3.15.0.314)
    nokogiri (1.6.7.2)
      mini_portile2 (~> 2.0.0.rc2)
    pry (0.10.3)
      coderay (~> 1.1.0)
      method_source (~> 0.8.1)
      slop (~> 3.4)
    puma (3.1.0)
    rabl (0.11.8)
      activesupport (>= 2.3.14)
    rack (1.6.4)
    rack-cors (0.4.0)
    rack-test (0.6.3)
      rack (>= 1.0)
    rake (10.5.0)
    roda (2.10.0)
      rack
    rspec (3.4.0)
      rspec-core (~> 3.4.0)
      rspec-expectations (~> 3.4.0)
      rspec-mocks (~> 3.4.0)
    rspec-core (3.4.2)
      rspec-support (~> 3.4.0)
    rspec-expectations (3.4.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.4.0)
    rspec-mocks (3.4.1)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.4.0)
    rspec-support (3.4.1)
    sequel (4.31.0)
    slop (3.6.0)
    thread_safe (0.3.5)
    tilt (2.0.2)
    timecop (0.8.1)
    transflow (0.3.0)
      dry-pipeline
      kleisli
      wisper
    tzinfo (1.2.2)
      thread_safe (~> 0.1)
    wisper (1.6.1)
    xpath (2.0.0)
      nokogiri (~> 1.3)

PLATFORMS
  ruby

DEPENDENCIES
  analytics-ruby
  bcrypt-ruby
  byebug
  capybara
  database_cleaner
  dotenv (~> 2.1, >= 2.1.1)
  dry-component!
  dry-data
  dry-validation!
  faraday (~> 0.9.2)
  faraday_middleware (~> 0.10.0)
  fog-aws
  levenshtein
  mime-types (= 3.0)
  mysql2
  newrelic_rpm (~> 3.15, >= 3.15.0.314)
  pry
  puma (~> 3.1)
  rabl
  rack-cors
  rack-test
  rake
  rodakase!
  rom!
  rom-mapper!
  rom-repository!
  rom-sql!
  rom-support!
  rspec
  timecop
  transproc!

BUNDLED WITH
   1.12.5
```

If you need any further information, feel free to ask me.
Thanks!

**Comment by flash-gordon** [comment]
@mauricioklein schema definition available starting from rom 2.0, and you're using https://github.com/rom-rb/rom/commit/ed18a8299198 which is rather outdated, try to use it from rubygems
@solnic I wonder why is that commit ^ tagged with `v2.0` o_O?
